### ISSUE TITLE
How to do global animation in React?

### ISSUE THREAD

**Issue by mengxi-ream** [VisActor/VChart#3219]
I tried global animation in this project: https://visualgallery.mengxi.work/chart/global-animation

this is what exactly I want: https://visactor.io/vchart/demo/storytelling/bar-to-pie

codebase: https://github.com/mengxi-ream/visual-gallery/blob/main/pages/chart/components/global-animation.tsx

However, I am not sure how to properly utilize `registerMorph` and `updateSpec` in React. I attempted to implement them, but encountered bugs, as demonstrated in the link provided earlier.

Here is my code:

```tsx
"use client";

import { VChart } from "@visactor/react-vchart";
import { IPieChartSpec, registerMorph } from "@visactor/vchart";
import { useTheme } from "nextra-theme-docs";
import { useState, useEffect } from "react";

const pieSpec: IPieChartSpec = {
  type: "pie",
  data: [
    {
      values: [
        { type: "1", value: Math.random() },
        { type: "2", value: Math.random() },
        { type: "3", value: Math.random() },
      ],
    },
  ],
  outerRadius: 0.8,
  innerRadius: 0.6,
  valueField: "value",
  categoryField: "type",
  tooltip: {
    visible: false,
  },
};

const barSpec = Object.assign({}, pieSpec, {
  type: "bar",
  xField: "type",
  yField: "value",
  seriesField: "type",
});

const specs = [pieSpec, barSpec];

export default function GlobalAnimation() {
  const { theme } = useTheme();
  console.log("theme", theme);
  const [specIndex, setSpecIndex] = useState(0);

  useEffect(() => {
    registerMorph();
    setInterval(() => {
      setSpecIndex((specIndex + 1) % specs.length);
    }, 2000);
  }, [specIndex]);

  return <VChart spec={specs[specIndex]} />;
}
```

To reproduce:
1. clone the project `git clone https://github.com/mengxi-ream/visual-gallery.git`
2. `pnpm i`
3. `pnpm dev`

**Comment by xile611** [comment]
@mengxi-ream the morphing animaiton of chart has been disabled

<img width="727" alt="image" src="https://github.com/user-attachments/assets/cd539ad3-d3b3-4169-9d03-744d927d31e8">

because there are too many useless update in `react-vchart`

**Comment by mengxi-ream** [comment]
> @mengxi-ream the morphing animaiton of chart has been disabled
> 
> <img alt="image" width="727" src="https://private-user-images.githubusercontent.com/6111424/367480818-cd539ad3-d3b3-4169-9d03-744d927d31e8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjYyODI2NTgsIm5iZiI6MTcyNjI4MjM1OCwicGF0aCI6Ii82MTExNDI0LzM2NzQ4MDgxOC1jZDUzOWFkMy1kM2IzLTQxNjktOWQwMy03NDRkOTI3ZDMxZTgucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDkxNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA5MTRUMDI1MjM4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MGQzMWFiZGU4Yjk4ODRkNTZiYTZiMzhlMGVlMDJjOTQ0ZjM0NmQ1ZDY5ZjAyYTc4YjcwMmNkYjA1M2Y0NWZhZSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.IvPYYxQLLkjwfsOEA94G8St5vqRYeb6isdoEPZfmnhE">
> because there are too many useless update in `react-vchart`

Does this mean in version `1.12.6` I just write something like `<VChart spec={spec} morphConfig?: {morph: true}` to turn morphing animation on?

**Comment by xile611** [comment]
> > @mengxi-ream the morphing animaiton of chart has been disabled
> > <img alt="image" width="727" src="https://private-user-images.githubusercontent.com/6111424/367480818-cd539ad3-d3b3-4169-9d03-744d927d31e8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjYyODI2NTgsIm5iZiI6MTcyNjI4MjM1OCwicGF0aCI6Ii82MTExNDI0LzM2NzQ4MDgxOC1jZDUzOWFkMy1kM2IzLTQxNjktOWQwMy03NDRkOTI3ZDMxZTgucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDkxNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA5MTRUMDI1MjM4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MGQzMWFiZGU4Yjk4ODRkNTZiYTZiMzhlMGVlMDJjOTQ0ZjM0NmQ1ZDY5ZjAyYTc4YjcwMmNkYjA1M2Y0NWZhZSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.IvPYYxQLLkjwfsOEA94G8St5vqRYeb6isdoEPZfmnhE">
> > because there are too many useless update in `react-vchart`
> 
> Does this mean in version `1.12.6` I just need to write something like `<VChart spec={spec} morphConfig: {morph: true} />` to turn morphing animation on?

yes

**Comment by mengxi-ream** [comment]
> > > @mengxi-ream the morphing animaiton of chart has been disabled
> > > <img alt="image" width="727" src="https://private-user-images.githubusercontent.com/6111424/367480818-cd539ad3-d3b3-4169-9d03-744d927d31e8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjYyODI2NTgsIm5iZiI6MTcyNjI4MjM1OCwicGF0aCI6Ii82MTExNDI0LzM2NzQ4MDgxOC1jZDUzOWFkMy1kM2IzLTQxNjktOWQwMy03NDRkOTI3ZDMxZTgucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDkxNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA5MTRUMDI1MjM4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MGQzMWFiZGU4Yjk4ODRkNTZiYTZiMzhlMGVlMDJjOTQ0ZjM0NmQ1ZDY5ZjAyYTc4YjcwMmNkYjA1M2Y0NWZhZSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.IvPYYxQLLkjwfsOEA94G8St5vqRYeb6isdoEPZfmnhE">
> > > because there are too many useless update in `react-vchart`
> > 
> > 
> > Does this mean in version `1.12.6` I just need to write something like `<VChart spec={spec} morphConfig: {morph: true} />` to turn morphing animation on?
> 
> yes

look forward to trying version 1.12.6. 😍

**Comment by xile611** [comment]
fixed in 1.12.6

**Comment by mengxi-ream** [comment]
@xile611 the code is not included in 1.12.6 (I've checked the release package)
